pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -321,9 +321,7 @@ type replicaStatus struct {
 func computeReplicaStatus(pods []*v1.Pod, minReadySeconds int32, currentRevision, updateRevision *apps.ControllerRevision) replicaStatus {
 	status := replicaStatus{}
 	for _, pod := range pods {
-		if isCreated(pod) {
```

Comment:
> Upon receiving the TERM signal, if we change the exit code to "0" (success) we don't see the issue with partition 1 i.e. the pod-0 container image remains the old one.

Could you put some reproduce steps at here? i have try it with preStop and it can reproduce this problem.

```yaml
          preStop:
            exec:
              command:
                - /bin/sh
                - -c
                - exit 0
```

Remind me if i missed something,Thanks!

Reply:
since you are able to reproduce it with prestop also.
maybe with prestop also the pod exit status is failed since the SIGTERM signal is not handled explicitly inside the pod container, by using the "command" and handling the "trap" I'm able to reproduce the problem.
 
 sts:   
 
 replicas: 2
 podManagementPolicy: "Parallel"
  updateStrategy:
    type: RollingUpdate
    rollingUpdate:
      partition: 1
  
  ...
    spec:
      restartPolicy: Always
      terminationGracePeriodSeconds: 15
      containers:
      - name: nginx
        image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
        imagePullPolicy: {{ .Values.image.pullPolicy }}	
        command:
         - "/bin/sh"
         - "-c"
         - |
           running=true
           **trap 'echo $(date +"%Y-%m-%d %H:%M:%S") - "Received SIGTERM signal. Exiting gracefully..."; sleep 1 ;running=false; ' TERM**
           # For example, an infinite loop to simulate a running process
              while $running; do
                 echo "$(date +'%Y-%m-%d %H:%M:%S') - Script is running..."
                 sleep 1
              done
              if ! $running; then
                 echo -e "$(date +'%Y-%m-%d %H:%M:%S') - exiting "
                 sleep 2
                 **exit  143 #$EXIT_CODE**
              fi
       
   here we can set the EXIT_CODE as 0 success, 143 gracefully shutdown or non-zero value.
   let me know if you need the sts deployment file and elaborate steps to reproduce it.

   Appreciate your efforts.